### PR TITLE
Add new 'common solution' for S_TYPExxx macros

### PR DESCRIPTION
**Pull request by MikeFultonDev**
- fix broken link to 'Updating docs', 
- add back 'Common Solutions'
- add the new common solution to working around broken S_TYPExxxx macros

### CODE REVIEW

File: docs/Guides/CommonSolutions.md [hunk 1]
```diff
@@ -144,3 +144,49 @@ cp /usr/lpp/cbclib/xlclang/etc/xlclang.cfg /u/fred/xlclang.cfg.zos204.noceertfm
 export CLC_CONFIG=/u/fred/xlclang.cfg.zos204.noceertfm
 ```
 
+## S_TYPEISSHM macro gives an error when compiled
+
+The error might be something like: `called object type 'int' is not a function or function pointer`
+
+This is due to a bug in the LE header file `sys/modes.h` which defines the macros as follows:
```
Comment (IgorTodorovskiIBM) [inline docs/Guides/CommonSolutions.md:151]: Can we fix this in zoslib by overriding sys/modes.h?

File: docs/_sidebar.md [hunk 2]
```diff
@@ -7,6 +7,7 @@
   - [Available ports](/Latest.md)
     - [Current Status](/Progress.md)
   - [FAQ](/Guides/FAQ.md)
+  - [Common Solutions](/Guides/CommonSolutions.md)
```
Comment (IgorTodorovskiIBM) [inline docs/_sidebar.md:10]: Oops, that was probably my fault

File: docs/Guides/CommonSolutions.md (hunk 1 above)
Comment (MikeFultonDev) [reply docs/Guides/CommonSolutions.md:151]: If we could fix this in zoslib that would be great.

**Comment by MikeFultonDev** [comment]
@IgorTodorovskiIBM ok to merge?

**Review by IgorTodorovskiIBM — APPROVED** [review]
LGTM

**Pull request merged**